nano/node/json_handler.cpp

```diff
@@ -911,18 +912,24 @@ void nano::json_handler::accounts_balances ()
 void nano::json_handler::accounts_representatives ()
 {
 	boost::property_tree::ptree representatives;
-	for (auto & accounts : request.get_child ("accounts"))
+	for (auto & account_from_request : request.get_child ("accounts"))
 	{
-		auto account (account_impl (accounts.second.data ()));
-		auto transaction (node.store.tx_begin_read ());
-		auto info (account_info_impl (transaction, account));
-
+		boost::property_tree::ptree entry;
+		auto account = account_impl (account_from_request.second.data ());
 		if (!ec)
 		{
-			boost::property_tree::ptree entry;
-			entry.put ("", info.representative.to_account ());
-			representatives.push_back (std::make_pair (accounts.second.data (), entry));
+			auto info = account_info_impl (node.store.tx_begin_read (), account);
+			if (!ec)
+			{
+				entry.put ("", info.representative.to_account ());
+				representatives.push_back (std::make_pair (account_from_request.second.data (), entry));
+				continue;
+			}
 		}
+		// Returns only the first error.
```

Comment:
What does this comment mean?
I cannot understand its meaning.

Reply:
there might be an error for the provided account number (invalid number), and also an error for when it is not found "Account not found". Instead of retrieving a list of errors this implementation only returns the first it finds.

Reply:
There cannot be a list of errors, there are only 2 errors and they cannot happen together

Reply:
Yes, returning a list was just I thought I had, it doesn't make sense for this function. My other explanation is: there is a stack where the account number is verified first and a new object is constructed for it, if there is an error on it, it won't look for its representatives and will return "Bad account number". But if it passes the first check and construct an account object, this account may still not be found. The first or the "only" error possible to be found is going to be added to the response.

Reply:
I must say, I still do not understand what it means... but I don't want to focus on it too much.
Can we add some unit tests instead to clearly show what the expected behaviour is?

Reply:
Removed the comment.